src/bucket/BucketManagerImpl.cpp

```diff
@@ -351,6 +351,13 @@ BucketManagerImpl::assumeState(HistoryArchiveState const& has)
         mBucketList.getLevel(i).setNext(has.currentBuckets.at(i).next);
     }
     mBucketList.restartMerges(mApp);
+    for (uint32_t i = 0; i < BucketList::kNumLevels; ++i)
```

Comment:
1. why do we need to do this?
2. if it does, I think you may have a race as this is basically racing with the merges that `restartMerges` started